docs: start rewriting smart router docs

andrew-stark-cko opened the pull request:
<!--- Ensure the title above contains the Jira issue name e.g PROCESS0UT-1 -->

## Description

The Smart Router API docs have some places where information gets repeated unnecessarily (eg, creation of an invoice) and sometimes details of a task are not in the obvious place (eg, client-side capture belongs in a page with server-side capture). Also, the docs could be improved in terms of clarity and detail.

## Solution

Rewrite the "Accepting card details" page to provide an overview but remove duplicated detail that is explained in other pages. Add a new page about creating an invoice so that the info has a "home" that is easily linked from elsewhere. Move the client-side capture details to "Capturing a payment" and generally rewrite the page text for clarity. Rename "Web browser payments" and "Mobile app payments" with "Tokenising...", since that is what those pages are essentially about.

## Notes

The tokenising web/mobile pages have not been rewritten yet but this will follow in a separate PR, and there will be further PRs for the other pages in the Smart Router API section.

I've tried removing some of the boilerplate code from the code samples (creating the ProcessOut object with the access tokens, etc) since this is now covered in its own page. Let me know if you think it is important to have these in every code sample, but to me, it didn't seem to make the samples clearer (ie, it makes it look as though the sample code directly follows the `import ProcessOut` and `client = new ProcessOut(...)` lines).

I've also started removing some of the Ruby partial dependencies from the pages, which leaves them as just ordinary Markdown. We can probably tidy this up later using template/webcomponent elements, etc, if necessary but all of this might be changed anyway if we move over to Checkout's doc systems.

## Jira Issue
<!--- Please attach a link to the Jira issue where possible. -->

jeremy-lejoux-cko commented on source/payments/payments-and-sca.md:
```diff
@@ -0,0 +1,84 @@
+the cart and, most importantly, the amount the customer will be charged.
+The flow of the transaction then continues as follows:
+
+1.  **Creating an invoice**
```
I think it would make sense to explain a bit more what is an Invoice in our system (i.e. how is it different from a Transaction object)

jeremy-lejoux-cko commented on source/payments/payments-and-sca.md:
```diff
@@ -0,0 +1,84 @@
+start the payment. The request will include the details of 
+the cart and, most importantly, the amount the customer will be charged.
+The flow of the transaction then continues as follows:
+
```
Before listing the flow itself below, I think it would make sense to list the "resources" that are required to perform a payment request on ProcessOut. For instance:

- A Customer object
- An Invoice
- A source that can be used to request funds (Card or existing token)

jeremy-lejoux-cko commented on source/payments/payments-and-sca.md:
```diff
@@ -0,0 +1,84 @@
+    > [tokenizing a card in a mobile app](/payments/mobile-sdks-reference/),
+    > and [saving a token to capture future payments](/payments/save-token-and-capture-later/)
+
+1.  **Capturing a payment**
```
This step should be focus on "Authorizing an payment". 
It might be misleading for merchants to focus on capturing here

jeremy-lejoux-cko commented on source/payments/payments-and-sca.md:
```diff
@@ -0,0 +1,84 @@
+    (CITs), your client-side code makes an API call to capture
+    the payment, using the token and invoice ID it received from the server. These
+    transactions are considered relatively high-risk, so capturing
+    on the client-side is recommended for maximum security.
```
For CIT we must say that in order to be SCA compliant, payments authorization request should happen in the front-end.

Capture should then be done in the backend though.

jeremy-lejoux-cko commented on source/payments/payments-and-sca.md:
```diff
@@ -0,0 +1,84 @@
+    transactions are considered relatively high-risk, so capturing
+    on the client-side is recommended for maximum security.
+
+    For [**Merchant Initiated Transactions**](https://www.tokenex.com/blog/ab-what-is-a-merchant-initiated-transaction-and-why-is-it-considered-low-risk#:~:text=A%20merchant%2Dinitiated%20transaction%20is,needed%20to%20make%20a%20payment.)
```
Here I think we should say that by nature, MIT transactions cannot be authorized in the front-end as the customer is not present. 

Hence why they should always be coming server-to-server with an existing (Stored) token

jeremy-lejoux-cko commented on source/payments/payments-and-sca.md:
```diff
@@ -0,0 +1,84 @@
+
+1.  **Listening for updates**
+
+    After capturing, you may find it useful to receive
```
As we are sending webhooks for every step (even before a successful authorization) I believe we should say something like:
> At any step of a transaction lifecycle, you might want to build custom logic based on events. ProcessOut sends webhooks for any authorization attempt and future events [...]

jeremy-lejoux-cko commented on source/payments/creating-invoice.md:
````diff
@@ -0,0 +1,122 @@
+```shell
+curl -X POST https://api.processout.com/invoices \
+    -u test-proj_gAO1Uu0ysZJvDuUpOGPkUBeE3pGalk3x:key_sandbox_mah31RDFqcDxmaS7MvhDbJfDJvjtsFTB \
+    -d name="Awesome invoice" \
````
Can we replace the placeholder with `Merchant internal oder ID`?

jeremy-lejoux-cko commented on source/payments/creating-invoice.md:
```diff
@@ -0,0 +1,122 @@
+title: Creating an invoice
+---
+
+# Creating an invoice
```
I think here we might want to guide the merchant a bit more on when to create an invoice within his payment flow.

Invoices are required to:

- List existing / possible alternative payment methods for a given invoice
- Initiate a payment request
- Storing information about an order within ProcessOut

Typically, it's during the invoice creation that merchants will send their internal IDs. That means that they should keep in mind that whenever they create the Invoice on ProcessOut they need to already have access to that data on their side.

jeremy-lejoux-cko commented on source/payments/capture-a-payment.md:
```diff
@@ -0,0 +1,381 @@
+[accepting card payments](/payments/payments-and-sca/)
+for an overview of the payment flow from invoice to capture.)
+
+You can capture a payment on the client-side using a
```
Captures can only happen from the backend as they require the private key in the request :)

jeremy-lejoux-cko commented on source/payments/capture-a-payment.md:
```diff
@@ -0,0 +1,381 @@
+title: Capturing a payment
+---
+
+# Capturing a payment
```
I think this step overall needs some breakdown between authorization & capture. 
Happy to discuss it in more details but basically:

- Authorization is always required to then initiate a capture
- Captures will always be coming from backend (except if the auto-capture is set)
- Authorization must come from front-end for CIT and from backend for MIT

Let's discuss this part :)

andrew-stark-cko commented:
@jeremy-lejoux-cko I've rewritten this stuff quite extensively now with more detail. Also, I think (hopefully!) that I've now understood the separate stages of authorisation and capture correctly. I've added a new page specifically about authorisation rather than discuss it on the page about capturing payment.

Hopefully, the text should now be a lot closer to the mark than the previous attempt. I'm still happy to chat about this if you think I'm still not getting it right but if there are only minor mistakes then it might be quicker to just respond with PR comments as usual.

michal-bielecki-cko commented on source/payments/authorizing-payment.md:
```diff
@@ -0,0 +1,251 @@
+payment and then reserve that money (ie, stop it from being used to
+pay for anything else). It is only when you
+*capture* the payment in a separate stage that the money actually
+gets transferred. Note that you can capture a payment up to 7
```
The actual money flow is not happening here. Acquiring bank will settle this with issuing bank later, merchant will also not receive the transfer instantly, but after a few days. I would change this to something like: "It is only when you *caputre* the payment you confirm and book the transaction."

andrew-stark-cko replied:
Added this and also mentioned that the transfer might take a few days.

jeremy-lejoux-cko commented on source/payments/payments-and-sca.md:
```diff
@@ -0,0 +1,188 @@
+customer with updates about receiving the order and despatching the
+goods.
+
+Another example of a payment process is a "buy now, pay later" purchase where
```
I would not mention "buy now pay later" here as this usually refers to specific payment method.
I'd rather refer to something a bit different like "Store card details for future purchase" or "subscription usecases"

jeremy-lejoux-cko commented on source/payments/payments-and-sca.md:
```diff
@@ -0,0 +1,188 @@
+    on an invoice directly but if you do so then the token cannot be reused
+    for future purchases.
+    An alternative approach is to use the card token to create a
+    [`Customer`](/refs/#customers) object that stores the card details along
```
I think the link is wrong here, should be : https://docs.processout.com/refs/#tokens
+ `Customer object` should be `Customer token`

jeremy-lejoux-cko commented on source/payments/authorizing-payment.md:
```diff
@@ -0,0 +1,252 @@
+This is necessary to comply with EU legislation on
+[Strong Customer Authentication (SCA)](/security-standards/#eu-payment-services-directive-2).
+
+[*Merchant Initiated Transactions*](https://www.tokenex.com/blog/ab-what-is-a-merchant-initiated-transaction-and-why-is-it-considered-low-risk#:~:text=A%20merchant%2Dinitiated%20transaction%20is,needed%20to%20make%20a%20payment.)
```
I think we should specify that in order for an MIT transaction to be approved banks will usually require an initial CIT transaction with 3DS to be done before.

jeremy-lejoux-cko commented:
Thanks a lot @andrew-stark-cko !! Looks more and more better :) 🎉

andrew-stark-cko commented:
@jeremy-lejoux-cko Great feedback, thanks! I've implemented what you've suggested. Please let me know if there is anything else to update or approve if it's in good shape now.

andrew-stark-cko commented on source/payments/authorizing-payment.md:
```diff
@@ -0,0 +1,254 @@
+This is necessary to comply with EU legislation on
+[Strong Customer Authentication (SCA)](/security-standards/#eu-payment-services-directive-2).
+
+[*Merchant Initiated Transactions*](https://www.tokenex.com/blog/ab-what-is-a-merchant-initiated-transaction-and-why-is-it-considered-low-risk#:~:text=A%20merchant%2Dinitiated%20transaction%20is,needed%20to%20make%20a%20payment.)
```
Good suggestion. I've fixed these here but I'll do a sweep through the docs to find all of these in a separate PR.

andrew-stark-cko commented on source/payments/authorizing-payment.md:
```diff
@@ -0,0 +1,254 @@
+(ie, capturing the payment will implicitly authorize it first).
+However, you can still authorize explicitly if you want to perform checks
+first and then capture later. Note that banks will usually only approve
+an MIT if it was initially set up with a CIT (which requires client-side
```
I know this seems counter-intuitive but it is standard practice to choose "a" or "an" based on the *sound* of the first letter of the abbreviation (as if you were saying it to yourself in your mind). If the first letter sounds as though it starts with a vowel, then you use "an" (so here, "MIT" sounds like "emm-eye-tee"). It's documented [here](https://editorsmanual.com/articles/indefinite-article-a-an-with-abbreviations/#a-an) with some examples.

andrew-stark-cko commented on source/payments/authorizing-payment.md:
```diff
@@ -0,0 +1,254 @@
+[Strong Customer Authentication (SCA)](/security-standards/#eu-payment-services-directive-2).
+
+[*Merchant Initiated Transactions*](https://www.tokenex.com/blog/ab-what-is-a-merchant-initiated-transaction-and-why-is-it-considered-low-risk#:~:text=A%20merchant%2Dinitiated%20transaction%20is,needed%20to%20make%20a%20payment.)
+(MITs) are authorized from the server-side. This is an
```
I've fixed this here but there are probably other examples elsewhere in the docs. I'll fix them up when I do the links to Checkout's docs.

andrew-stark-cko commented on source/payments/authorizing-payment.md:
```diff
@@ -0,0 +1,254 @@
+### Client-side authorization
+
+You must already have an [`Invoice`](/payments/creating-invoice/) ID
+and a *payment source* (a [card token](/payments/processoutjs-reference/)
+or a [customer token](/payments/save-token-and-capture-later/))
```
Done.

andrew-stark-cko commented on source/payments/authorizing-payment.md:
````diff
@@ -0,0 +1,254 @@
+
+```swift
+var authorizationRequest = AuthorizationRequest(
+    source:"card_1jSEVrx7oaRta1KEdxoMWbiGkK2MijrZ",
````
Done.

andrew-stark-cko commented on source/payments/authorizing-payment.md:
````diff
@@ -0,0 +1,254 @@
+```
+
+```go
+tr, _ := iv.Authorize("card_1jSEVrx7oaRta1KEdxoMWbiGkK2MijrZ")
````
Done.

andrew-stark-cko commented on source/payments/capture-a-payment.md:
```diff
@@ -0,0 +1,146 @@
+When the customer has authorized payment from their card,
+the final step is to *capture* the payment on the server. This
+involves making a request to a Payment Service Provider (PSP) to
+complete the transfer of money from the customer's bank to
```
Done.

andrew-stark-cko commented on source/payments/capture-a-payment.md:
```diff
@@ -0,0 +1,146 @@
+if the payment is successful or an error code or null value if it isn't.
+The `Transaction` has a `status` attribute that gives more detail about the
+exact stage it has reached. A successful payment will usually have a
+`status` value of `completed` but there are a number of other useful status
```
Done.

andrew-stark-cko commented on source/payments/capture-a-payment.md:
```diff
@@ -0,0 +1,146 @@
+The order is often complete after you have captured the payment,
+but in some cases its status can still change after capture.
+We recommend that you stay updated about any status changes
+by [using webhooks to respond to events](/payments/handle-status-changes-webhooks/)
```
Done.

andrew-stark-cko commented on source/payments/creating-invoice.md:
````diff
@@ -0,0 +1,142 @@
+```shell
+curl -X POST https://api.processout.com/invoices \
+    -u test-proj_gAO1Uu0ysZJvDuUpOGPkUBeE3pGalk3x:key_sandbox_mah31RDFqcDxmaS7MvhDbJfDJvjtsFTB \
+    -d name="Merchant internal order ID" \
````
Done, and I'll make sure this is removed elsewhere too. (I was just following an earlier suggestion from Jeremy about this but we should remove it if it's not good practice.)

andrew-stark-cko commented on source/payments/payments-and-sca.md:
```diff
@@ -0,0 +1,188 @@
+    (or recurring payments from a customer) by storing their contact and
+    payment details.
+
+-   A communication system that uses [webhooks](https://www.webhooks.org/)
```
Done.

andrew-stark-cko commented on source/payments/payments-and-sca.md:
```diff
@@ -0,0 +1,188 @@
+    required in your workflow.
+     
+    For example, after capturing, you may find it useful to receive
+    updates from the PSP about the status of the payment. For card
```
Done.

andrew-stark-cko commented on source/payments/payments-and-sca.md:
```diff
@@ -0,0 +1,188 @@
+    updates from the PSP about the status of the payment. For card
+    payments, the PSP will confirm success or failure
+    synchronously as a response to the request. However, a customer
+    might later obtain a
```
Done (and I'll also change this elsewhere if I find it).
